### DISCUSSION TITLE
Auto Pilot Now Different When We Log In

### DISCUSSION THREAD

**Discussion by BonzoDogDooDah** [lightninglabs/lightning-terminal#755] [Q&A]
Hi All,

We have Auto Pilot enabled and when we first set it up it showed that you could enable or disable a channnel so we chose to enable all of them. We now have enough channels that we have been accepted as a +Good node and are no longer pending, thank you. We would like to know why when we now log in and choose Auto Pilot we can no longer choose to disable a channel from Auto Pilot if we decide to do so? Actually no options show up at all now? Also is it usual for the Auto Pilot to lower most of our channels to 1 and then the rest to very low fee rates? Our goal is to help the network but to also get some return on our investment if possible? If you have any suggestions as to how we may reach these goals please let us know. Our node is named: Handshake. Thank you :)

**Comment by bitromortac** [comment]
You should be able to enable/disable channels for autofees in the Loop view. Each channel has a switch where you can toggle autofees. If you see small fee rates, then you're either not providing liquidity where it's needed, or you don't have inbound liquidity in channels which might want to send over your node. You can start rebalancing/looping out channels that are inbound-depleted, you could select them by analyzing your past forwards.

**Comment by BonzoDogDooDah** [comment]
Hi,

Thank you for your reply. Can we set up our node in such a way that it will self rebalance itself without having to loop out or do circular rebalancing as we have found those options to be cost prohibitive due to the fact that we are not able to gain that investment back after the channel gets refilled with inbound liquidity. Please let us know what we can do? Thank you. :)

-- 
 Sent with Tuta; enjoy secure & ad-free emails: 
 https://tuta.com


May 6, 2024, 07:13 by ***@***.***:

**Reply by bitromortac** [reply]
Unfortunately this is a trial and error (and market) process, which we try to automate in the future, but this is not there yet. You'll need to experiment and sometimes also invest more in order to find out potential sources of income. It would also be important to raise fees to targets where you'd expect funds to flow after increasing inbound liquidity (autofees doesn't interact with loop/rebalancing yet). This way you may be able to get some of the invested fees back.

**Comment by utente9382** [comment]
How private is it to use the autofees? Will lightning labs server see all my utxos and lightning balances?

**Reply by racarr202** [reply]
It's time to dig into the source code I think to confirm lightning labs is not pulling a fastie on us. 

As a side note, if you were concerned about privacy, you can always considder using [LNDg ](https://github.com/cryptosharks131/lndg) as its autofees is all managed locally.

**Reply by utente9382** [reply]
I am aware of LNDg, but my hardware is weak for it.

**Reply by bitromortac** [reply]
Autofees uses these RPCs `/lnrpc.Lightning/ForwardingHistory` `/lnrpc.Lightning/UpdateChannelPolicy` `/lnrpc.Lightning/ListChannels` `/lnrpc.Lightning/FeeReport` and doesn't have access to any wallet related information or other calls. The amounts in `ListChannels` (channel balances) and `ForwardingHistory` are obfuscated somewhat in order for the algorithm to still work (it requires balances with some accuracy to determine if a channel is depleted). The obfuscation is also applied to other fields in responses to these RPCs.

You can read more about it here: https://docs.lightning.engineering/lightning-network-tools/lightning-terminal/autofees, and the source is here: https://github.com/lightninglabs/lightning-terminal/blob/master/firewall/privacy_mapper.go.

We worked hard to balance the trade offs between absolute, perfect privacy and creating a useful service for end users. The most important part is to not correlate payment forwards, which we worked hard to make possible. We're actively working to improve the code in this area and messaging around this feature. We're open to feedback as always.

**Reply by utente9382** [reply]
Very good answer, my IQ80 doesn't allow me to verify the code. I will trust you on that for the moment. Thanks.
